design/open/acl/rbac-to-acl.md

```diff
@@ -0,0 +1,202 @@
+levels of granularity.
+
+### Upgrade to simplified
+
+On startup, lakeFS will check the KV schema version.  If it has not upgraded
+to ACLs then lakeFS will check whether an upgrade is required:
+
+* If the 4 basic policies have not been modified, and no other policies have
+  been defined, and no users have attached policies: no upgrade is required.
+  lakeFS will merely upgrade the KV schema version.
+* Otherwise lakeFS will not start until an upgrade script runs.
+
+#### The upgrade script
+
+Admins will run the upgrade script if required to do so by lakeFS or if they
+want to be very sure (for instance, there is a minor race condition if there
+is more than one concurrent lakeFS).  We force admins to try the script on a
+dry run using a `--yes` flag: without it, the script only prints the planned
+changes.
+
+The upgrade script will _ensure_ that the 4 default global groups exist, and
+_modify_ existing groups to fit into the new ACLs scheme.
+
+* When creating the 4 default global groups: if another group exists and has
+  the desired name, upgrading will rename it by appending ".orig".  So after
+  upgrading the 4 default global groups exist, with these known names.
+* For any group, upgrading configured policies follows these rules, possibly
+  **increasing** access:
+
+  1. Any "Deny" rules are stripped, and a warning printed.
+  1. "Manage own credentials" is added.
+  1. If any actions outside of "fs:" are allowed, the group becomes an Admin
+     group, a warning is printed, and no further changes apply.
+  1. The upgrade script *unifies* repositories: If a permission applies to a
+     set of repositories with a wildcard, permissions are unified to **all**
+     repositories.  Otherwise they apply to the list of all repositories, in
+     all the policies.
+  1. The upgrade script *unifies* actions: it selects the _least_ permission
+     of Read, Write, Super that contains all of the allowed actions.  If any
+     allowed action contains a wildcard this still makes sense.
+* The upgrade script prints the new permissions of the group.[^1]
+
+
+[^1] Even JSON-diff cannot be used report detailed changes, as many policies
+	 are unified into a single policy.
+
+### PBAC user experience with external authorization server
+
+If lakeFS is attached to an external authorization server, the existing PBAC
+GUI may continue to be used.  It is not possible to use both types of GUI at
+the same time: Moving from PBAC to simplified may only be performed once and
+**will** lose configuration.
+
+## Implementation
+
+lakeFS enforcement will remain backed by PBAC.  We shall add a "ui" field to
+policies.  It will contain a structure that backs the UI:
+
+<table>
+<tr><th>Read all repos</th><th>Read repos `foo`, `bar`</th></tr>
+<tr><td>
+```json
+{
+  "statement": [
+    {
+      "action": ["fs:List*", "fs:Read*"],
+      "effect": "allow",
+      "resource": "*"
+    }
+  ],
```

Comment:
I wonder, is the UI going to be aware of this configuration? or you placed it here for the example and comparison between what the server is aware of and taking care of the translation?

Reply:
The UI will use new APIs.  Those APIs will read and write (only) the UI part of the object.  OpenAPI will mirror the [ACL model in this PR](https://github.com/treeverse/lakeFS/pull/5297/files#diff-fa8533144f17ae3823634d8731349205bbe38a54930cf123241bca0c814ff412R153-R156).